frugalos_raft/src/storage/log.rs

```diff
@@ -150,3 +151,129 @@ impl Future for LoadLogInner {
         }
     }
 }
+
+/// Raft用のローカルログを削除するための`Future`実装.
+pub struct DeleteLog {
+    logger: Logger,
+    event_tx: mpsc::Sender<Event>,
+    future: BoxFuture<()>,
+}
+impl DeleteLog {
+    /// Creates a new `DeleteLog` instance.
+    pub(crate) fn new(handle: &Handle, event_tx: mpsc::Sender<Event>, node: LocalNodeId) -> Self {
+        let logger = handle.logger.clone();
+        let future = into_box_future(
+            handle
+                .device
+                .request()
+                .deadline(Deadline::Infinity)
+                .delete_range(node.to_available_lump_id_range())
+                .map(|_| ()),
+        );
+        info!(logger, "[START] DeleteLog");
+        Self {
+            logger,
+            event_tx,
+            future,
+        }
+    }
+}
+impl Future for DeleteLog {
+    type Item = ();
+    type Error = Error;
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        if let Async::Ready(()) = track!(self.future.poll())? {
+            info!(self.logger, "[FINISH] DeleteLog");
+            let _ = self.event_tx.send(Event::LogSuffixDeleted);
```

Comment:
note: On the completion of deletion, `Event::LogSuffixDeleted` is sent to `Storage::poll_and_handle_event`.